src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -141,13 +134,20 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
       const root : FiberRoot = (container.stateNode : any);
       const current = root.current;
 
-      root.pendingContext = getContextForSubtree(parentComponent);
+      const context = getContextForSubtree(parentComponent);
+      if (root.context === null) {
```

Comment:
@acdlite Keeping pendingContext on the root fiber seems fragile and doesn't work correctly with incremental. Can we move it to the update queue of the host root now?

On the other hand, this is all just a temporary hack until we can switch to portals.